### PR TITLE
ON-843: Deploy Nft Rental Marketplace Moonbeam

### PR DESCRIPTION
**Pull request by karacurt**

### CODE REVIEW

File: utils/deploy-upgradeable.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID || 'AKIAxxxxxxxxxxxxxxxx', // credentials for your IAM user with KMS access
+  secretAccessKey: process.env.AWS_ACCESS_KEY_SECRET || 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', // credentials for your IAM user with KMS access
+  region: 'us-east-1', // region of your KMS key
+  keyId: process.env.AWS_KMS_KEY_ID || 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', // KMS key id
+}
+const NETWORK = network.name as Network
+const networkConfig: any = network.config
+const provider = new ethers.JsonRpcProvider(networkConfig.url || '')
+const deployer = new AwsKmsSigner(kmsCredentials).connect(provider)
+export async function deployUpgradeableContract(
```
Comment (ernanirst) [inline utils/deploy-upgradeable.ts:17]:
```suggestion
const deployer = new AwsKmsSigner(kmsCredentials).connect(provider)

export async function deployUpgradeableContract(
```

File: scripts/orium-marketplace-royalties/01-deploy.ts [hunk 2]
```diff
@@ -1,89 +1,23 @@
...
-/* const FEE_DATA: any = {
-  maxFeePerGas: ethers.utils.parseUnits('80', 'gwei'),
-  maxPriorityFeePerGas: ethers.utils.parseUnits('50', 'gwei'),
-}
-provider.getFeeData = async () => FEE_DATA */
-const deployer = new AwsKmsSigner(kmsCredentials).connect(provider)
+const OPERATOR_ADDRESS = KMSDeployer.address
+const INITIALIZER_ARGUMENTS: string[] = [
+  OPERATOR_ADDRESS,
+  ERC7432WrapperForERC4907.address,
```
Comment (ernanirst) [inline scripts/orium-marketplace-royalties/01-deploy.ts:14]: Let's use `ZeroAddress` for future deployments. We won't have this contract on every network

File: utils/deploy-upgradeable.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        ],
+        name: 'transferOwnership',
+        outputs: [],
+        stateMutability: 'nonpayable',
+        type: 'function',
+      },
+    ]
+    const proxyAdminContract = new ethers.Contract(deploymentInfo[PROXY_CONTRACT_NAME].proxyAdmin, abi, deployer)
+    await proxyAdminContract.transferOwnership(OPERATOR_ADDRESS)
+    print(colors.success, `Proxy admin ownership transferred to: ${OPERATOR_ADDRESS}`)
```
Comment (ernanirst) [inline utils/deploy-upgradeable.ts:97]: Since the deployer is the owner, do we need to transfer ownership?

File: utils/ethers-aws-kms-signer.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
+import { GetPublicKeyCommand, KMSClient, SignCommand } from '@aws-sdk/client-kms'
```
Comment (ernanirst) [inline utils/ethers-aws-kms-signer.ts:1]: Put link to source file

**Review by ernanirst — CHANGES_REQUESTED** [review]

File: utils/deploy-upgradeable.ts (hunk 3 above)
Comment (karacurt) [reply utils/deploy-upgradeable.ts:98]: This is a generic script, just to make sure that the ownership is transferred to the operator passed as parameter. We cannot count that the operator will always be same address as the deployer

**Review by ernanirst — APPROVED** [review]

**Pull request merged**